angr/analyses/identifier/runner.py

```diff
@@ -75,10 +75,10 @@ def _get_recv_state(self):
             entry_state.unicorn.max_steps = 10000
 
             pg = self.project.factory.simgr(entry_state)
+            stop_addr = self.project._simos.syscall_from_number(2).addr
             num_steps = 0
             while len(pg.active) > 0:
-                syscall = self.project._simos.syscall_table.get_by_addr(pg.one_active.addr)
```

Comment:
Still going through the diff, so maybe I'll answer my own question in a sec, but is this the only way to check the type of snippet that's about to be executed? It'd be cool to have a more general interface (_eventually_), though I guess the way engines are designed might make that specific thing a bit tricky.